iptables/conntrack.go

```diff
@@ -0,0 +1,86 @@
+package iptables
+
+import (
+	"errors"
+	"net"
+	"os/exec"
+	"strings"
+	"sync"
+	"syscall"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/vishvananda/netlink"
+)
+
+var (
+	// ErrConntrackNotConfigurable means that conntrack module is not loaded or does not have the netlink module loaded
+	ErrConntrackNotConfigurable = errors.New("conntrack is not available")
+	isConntrackAvailable        bool
+	initConntrackOnce           sync.Once
+)
+
+const (
+	conntrackNetlinkModule = "nf_conntrack_netlink"
+)
+
+// IsConntrackProgrammable returns true if the handle supports the NETLINK_NETFILTER and the base modules are loaded
+func IsConntrackProgrammable(nlh *netlink.Handle) bool {
+	conntrackInitCheck()
+	return nlh.SupportsNetlinkFamily(syscall.NETLINK_NETFILTER) && isConntrackAvailable
```

Comment:
Given you attempt to create the netlink handle with the syscall.NETLINK_NETFILTER regardless:
```
+	fams := []int{syscall.NETLINK_ROUTE, syscall.NETLINK_NETFILTER}
```
this check here is redundant `nlh.SupportsNetlinkFamily(syscall.NETLINK_NETFILTER)` .